pip/utils/__init__.py

```diff
@@ -54,6 +55,37 @@ def get_prog():
     return 'pip'
 
 
+def default_user_site():
+    # Avoid circular import, the running_under_virtualenv should probably be
+    # in pip.utils anyways TBH.
+    from pip.locations import running_under_virtualenv, distutils_scheme
+
+    # If we're running inside of a virtual environment, we do not want to
+    # install to the --user directory.
+    if running_under_virtualenv():
+        return False
+
+    # If the Python we're running under does not have their user packages
+    # enabled then we do not want to install to the user directory since it
+    # may or may not work or exist.
+    if not site.ENABLE_USER_SITE:
+        return False
+
+    # If any of our potentional locations for writing files is not writable by
+    # us then we want to use the --user scheme instead of the --global scheme.
+    # TODO: We should figure out a way to make this work for --root and
+    #       --isolated and such options as well.
+    # TODO: Figure out if this works when the directories don't exist.
+    for path in distutils_scheme("").values():
+        if not os.access(path, os.W_OK):
```

Comment:
This doesn't seem to behave properly on Windows. `os.access(r'C:\Windows', os.W_OK)` is True for me, but `open(r'C:\Windows\test.file', 'w').close()` fails with errors on 2.7 and 3.4.

I was expecting this check to happen at a higher level - about the point where pip would terminate with an error but instead retry with `--user`.